src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -169,23 +169,17 @@ internal bool ShouldTraceStatement
         /// </summary>
         internal AutomationEngine Engine { get; private set; }
 
-        /// <summary>
-        /// Get the RunspaceConfiguration instance
-        /// </summary>
-        internal RunspaceConfiguration RunspaceConfiguration { get; }
-
         internal InitialSessionState InitialSessionState { get; }
 
         /// <summary>
-        /// True if the RunspaceConfiguration/InitialSessionState is for a single shell or false otherwise.
+        /// True if the InitialSessionState is for a single shell or false otherwise.
         /// </summary>
         ///
         internal bool IsSingleShell
         {
             get
             {
-                RunspaceConfigForSingleShell runSpace = RunspaceConfiguration as RunspaceConfigForSingleShell;
-                return runSpace != null || InitialSessionState != null;
+                return InitialSessionState != null;
```

Comment:
It looks to me it's always SingleShell now, as `InitialSessionState` is initialized in the constructor and cannot be null.
So, all `IsSingleShell` related the code can be cleaned as well.

Reply:
yup, will clean up